File: src/asset.py
from dataclasses import dataclass, field
from typing import Optional

@dataclass
class Cryptocurrency:
    ticker: str
    name: str
    circulating_supply: float
    max_supply: Optional[float] = None
    market_cap: float
    volume_24h: float
    change_24h: float 
    all_time_high: Optional[float] = None 
    all_time_low: Optional[float] = None

    @property
    def price(self) -> float:
        if self.circulating_supply > 0:
            return self.market_cap / self.circulating_supply
        else:
            return 0.0

File: WitsFrontend/Models/Asset.cs
using System.ComponentModel.DataAnnotations;
using Postgrest.Attributes;
using Postgrest.Models;

namespace WitsFrontend.Models;

[Table("Assets")]
public class Asset : BaseModel
{
    [PrimaryKey("id", false)]
    public int Id { get; set; }

    [Column("ticker")]
    [Required]
    public string Ticker { get; set; }

    [Column("name")]
    [Required]
    public string Name { get; set; }

    [Column("circulating_supply")]
    public decimal CirculatingSupply { get; set; }

    [Column("max_supply")]
    public decimal MaxSupply { get; set; }

    [Column("market_cap")]
    [Required]
    public decimal MarketCap { get; set; }

    [Column("price")]
    [Required]
    public decimal Price { get; set; }

    [Column("volume_24h")]
    [Required]
    public decimal Volume24h { get; set; }

    [Column("change_24h")]
    [Required]
    public decimal Change24h { get; set; }

    [Column("all_time_high")]
    public decimal AllTimeHigh { get; set; }

    [Column("all_time_low")]
    public decimal AllTimeLow { get; set; }
}
